integration/clean/main.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent/integration/clean/ami"
+	"github.com/aws/amazon-cloudwatch-agent/integration/clean/dedicated_host"
+	"github.com/aws/amazon-cloudwatch-agent/integration/clean/ssm"
+)
+
+const (
+	defaultDaysToKeep = 30
+	cleanerTypeAll    = "all"
+	delimiter         = ","
+)
+
+var (
+	// flags
+	daysToKeep    int
+	cleanersToRun string
+
+	cleanerTypes   = []string{ssm.Type, ami.Type, dedicated_host.Type}
+	cleanerOptions = strings.Join(cleanerTypes, delimiter)
+)
+
+func init() {
+	flag.IntVar(&daysToKeep, "keep", defaultDaysToKeep, "Days to keep a resource before cleaning it.")
+	flag.StringVar(&cleanersToRun, "clean", cleanerTypeAll,
+		fmt.Sprintf("Determines which cleaners to run. (e.g. -clean ami%sssm)\nOptions: %s,%s", delimiter, cleanerTypeAll, cleanerOptions))
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+
+	keepDuration := -1 * time.Hour * 24 * time.Duration(daysToKeep)
+
+	if strings.Contains(cleanersToRun, cleanerTypeAll) {
+		cleanersToRun = cleanerOptions
+	}
+
+	// The date used to determine if a resource can be cleaned.
+	// Anything older than the date will be removed.
+	expirationDate := time.Now().UTC().Add(keepDuration)
+	log.Printf("Expiration date set to %v", expirationDate)
+
+	for _, cleaner := range strings.Split(cleanersToRun, delimiter) {
+		switch cleaner {
+		case ami.Type:
+			if err := ami.Clean(ctx, expirationDate); err != nil {
+				log.Printf("%v", err)
+			}
+		case dedicated_host.Type:
+			if err := dedicated_host.Clean(ctx, expirationDate); err != nil {
+				log.Printf("%v", err)
+			}
+		case ssm.Type:
+			if err := ssm.Clean(ctx, expirationDate); err != nil {
+				log.Printf("%v", err)
+			}
```

Comment:
What bothers me about this is that it's added work to hook up a new cleaner here. For example, if we introduce k8s or ECS integration testing, we will probably want to add some form of cleaning up either the docker images, the clusters, or something - not sure what exactly yet. 

When we eventually add those integration tests, I wouldn't be surprised if we end up needing to add yet another switch statement here. It is fine in this scenario but makes it a little more cumbersome to add new things. 

Look at telegraf as an example. If someone contributes a new plugin to telegraf, the maintainers of telegraf don't have to also include a code change in the main code to accept that new plugin, as far as I know. If the point here is to consolidate and make it easier to use all of these different cleaner functions, we should treat them like individual cleaner "plugins" that we can use interchangeably. Adding the `Type` field to each cleaner should have been the first clue that this is a necessary change. We should not need to add a new switch statement here if we were to add a new cleaner.

Reply:
That's a good point of `telegraf`. Maybe I'm more biased regarding how `open-telemetry` framework. Even though the `open-telemetry` can treat these as individual cleaner "plugins"; however, it still provides the options for down-stream repo to consolidate their options instead of `telegraf` which only specify their options only to the specific functions. However, consider the small cleaner functions, I personally more favor to `open-telemetry` framework than towards `telegraf`. But It's good to have a new pair of eyes on this.